### PR TITLE
test(input-date-picker): add test for no longer closing date-picker when using arrow and page keys.

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #9082 

## Summary

No longer closes `date-picker` in `input-date-picker` when user interacts with arrowKeys (arrowDown/arrowUp) or pageKeys (pageUp/pageDown).

### CODE REVIEW

**Comment by anveshmekala** [comment]
should the PR title use `test` or `fix` considering the issue is resolved at date-picker PR #9063. We can use  https://github.com/Esri/calcite-design-system/issues/9082 to verify the fix for input-date-picker

**Comment by driskull** [comment]
@anveshmekala I'm only seeing changes to a test in this PR.

**Comment by driskull** [comment]
> should the PR title use test or fix considering the issue is resolved in date-picker with PR https://github.com/Esri/calcite-design-system/pull/9063. We can use https://github.com/Esri/calcite-design-system/issues/9082 to verify the fix for input-date-picker

Yes, it should use `test(input-date-picker)`

File: packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts [hunk 1]
```diff
@@ -1145,4 +1145,124 @@ describe("calcite-input-date-picker", () => {
...
+      await page.keyboard.press("Tab");
+      await page.waitForChanges();
+      await page.keyboard.press("ArrowDown");
+      await page.waitForChanges();
+      calendar = await page.find(`calcite-input-date-picker >>> .${CSS.calendarWrapper}`);
+      expect(await calendar.isVisible()).toBe(true);
+
+      await page.keyboard.press("PageDown");
+      await page.waitForChanges();
+      calendar = await page.find(`calcite-input-date-picker >>> .${CSS.calendarWrapper}`);
```
Comment (jcfranco) [inline packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts:1258]: This selector is used frequently, can you store it in a local variable and reuse?

File: packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts [hunk 2]
```diff
@@ -1145,4 +1145,124 @@ describe("calcite-input-date-picker", () => {
...
+
+      await page.keyboard.press("ArrowDown");
+      await page.waitForChanges();
+      calendar = await page.find(`calcite-input-date-picker >>> .${CSS.calendarWrapper}`);
+      expect(await calendar.isVisible()).toBe(true);
+
+      await page.keyboard.press("Enter");
+      await page.waitForChanges();
+      calendar = await page.find(`calcite-input-date-picker >>> .${CSS.calendarWrapper}`);
+      expect(await calendar.isVisible()).toBe(false);
```
Comment (jcfranco) [inline packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts:1196]: Could you add finer-grained assertions on the month navigation? It looks like this is only asserting that the calendar is displayed. Applies to the test below.

File: packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts [hunk 3]
```diff
@@ -1145,4 +1145,124 @@ describe("calcite-input-date-picker", () => {
...
+      await page.keyboard.press("PageDown");
+      await page.waitForChanges();
+      calendar = await page.find(`calcite-input-date-picker >>> .${CSS.calendarWrapper}`);
+      expect(await calendar.isVisible()).toBe(true);
+
+      await page.keyboard.press("Enter");
+      await page.waitForChanges();
+      calendar = await page.find(`calcite-input-date-picker >>> .${CSS.calendarWrapper}`);
+      expect(await calendar.isVisible()).toBe(false);
+      expect(await inputDatePicker.getProperty("value")).not.toEqual(["2024-01-01", "2024-02-10"]);
```
Comment (jcfranco) [inline packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts:1265]: For clarity, can you assert on the actual values instead? Assertion-wise, this and the assertion on L1237 would pass if they are the same.

**Review by jcfranco — CHANGES_REQUESTED** [review]
@anveshmekala Looking good!

File: packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts (hunk 3 above)
Comment (anveshmekala) [reply packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts:1265]: make sense.

**Review by jcfranco — APPROVED** [review]
# ⭐️⭐️⭐️⭐️⭐️
###### _Reviewed on 4/26/2024_

### **Awesome! Would review again!**

**Pull request merged**